File: leetcode/median-of-two-sorted-arrays.java
// https://leetcode.com/problems/median-of-two-sorted-arrays/submissions/

class Solution {
    // O(n+m) solution - should find O(log(n+m)) approach
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int[] merged = new int[nums1.length+nums2.length];
        int one = 0;
        int two = 0;
        
        for(int i = 0; i<merged.length; i++){
            if(one < nums1.length && two < nums2.length){
                merged[i] = nums1[one] <= nums2[two] ? nums1[one++] : nums2[two++];
            }else{
                merged[i] = one < nums1.length ? nums1[one++] : nums2[two++];
            }
        }
        
        for(int i = 0; i<merged.length; i++){
            System.out.println(merged[i]);
        }
        
        return merged.length % 2 == 0 ? (double)(merged[merged.length/2]+merged[merged.length/2-1])/2 : merged[merged.length/2];
    }
}

File: leetcode/median_of_two_sorted_arrays.py
# https://leetcode.com/problems/median-of-two-sorted-arrays


class Solution:
    def findMedianSortedArrays(self, nums1: list[int], nums2: list[int]) -> float:
        combined: list[int] = []
        index_one: int = 0
        index_two: int = 0

        while index_one < len(nums1) and index_two < len(nums2):
            if nums1[index_one] <= nums2[index_two]:
                combined.append(nums1[index_one])
                index_one += 1
            else:
                combined.append(nums2[index_two])
                index_two += 1

        if index_one < len(nums1):
            while index_one < len(nums1):
                combined.append(nums1[index_one])
                index_one += 1
        else:
            while index_two < len(nums2):
                combined.append(nums2[index_two])
                index_two += 1

        return (
            combined[(len(combined) // 2)]
            if len(combined) % 2 == 1
            else (combined[(len(combined) // 2)] + combined[(len(combined) // 2) - 1]) / 2
        )
